ginivision/src/main/java/net/gini/android/vision/review/ReviewActivity.java

```diff
@@ -321,6 +321,8 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
                 setResult(resultCode, data);
                 finish();
                 clearMemory();
+            } else {
```

Comment:
And then we check for that result code here and it would be clearer what our intent is I think:
```
            if (resultCode == AnalysisActivity.RESULT_NO_EXTRACTIONS) {
                finish();
                clearMemory();
            } else if (mBackButtonShouldCloseLibrary
                    || resultCode != Activity.RESULT_CANCELED) {
                setResult(resultCode, data);
                finish();
                clearMemory();
            }
```

Reply:
I see that `RESULT_NO_EXTRACTIONS` can makes sense [here](https://github.com/gini/gini-vision-lib-android/pull/70#discussion_r138323423) but then we have to override the back/up button behaviour as well. Those both set `RESULT_CANCELED` when used.

Reply:
They don't matter in this case, since on no extractions, the analysis screen is finished with `RESULT_NO_EXTRACTIONS`. If back/up is used, nothing should change.